flagsmith-engine/segments/evaluators.ts

```diff
@@ -55,18 +55,21 @@ function traitsMatchSegmentRule(
     );
 }
 
-function traitsMatchSegmentCondition(
+export function traitsMatchSegmentCondition(
     identityTraits: TraitModel[],
     condition: SegmentConditionModel,
     segmentId: number | string,
     identityId: number | string
 ): boolean {
-    if (condition.operator == PERCENTAGE_SPLIT) {
-        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(condition.value);
-    }
-
     const traits = identityTraits.filter(t => t.traitKey === condition.property_);
     const trait = traits.length > 0 ? traits[0] : undefined;
-
-    return trait ? condition.matchesTraitValue(trait.traitValue) : false;
-}
+    if (condition.operator == PERCENTAGE_SPLIT) {
+        return getHashedPercentateForObjIds([segmentId, identityId]) <= parseFloat(<string>condition.value);
```

Comment:
We shouldn't move this logic below the logic to extract the traits since we don't need the traits for this. It would mean that the application is doing unnecessary logic every time it's evaluating a percentage split segment.

Reply:
We also need to remove the addition of the `<string>` syntax in the call to the `parseFloat()` function.